subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/resolver/AbstractResourcePattern.java

```diff
@@ -84,4 +84,41 @@ protected String substituteTokens(String pattern, Map<String, String> attributes
         attributes.put(IvyPatternHelper.REVISION_KEY, componentIdentifier.getVersion());
         return attributes;
     }
+
+    @Override
+    public boolean isComplete(String organisation, String module) {
```

Comment:
⭕️ I think we should accept `ModuleIdentifier` & friends here instead of individual strings. That makes it much more obvious what this method is about.